### DISCUSSION TITLE
JVM parameter configuration failure

### DISCUSSION THREAD

**Discussion by MmringZee** [apache/rocketmq#7759] [Q&A]
I have changed the JVM related parameters of runserver.sh and runbroker.sh by modifying them, but with the command
`$ nohup sh bin/mqnamesrv &`  and  `$ nohup sh bin/mqbroker -n localhost:9876 --enable-proxy &` . After starting rocketmq, I found that rocketmq's memory footprint still reached nearly 8 GB.
 
![Uploading 20240117-184124.jpg…]()
![20240117-184132](https://github.com/apache/rocketmq/assets/113001598/f5863064-ae9a-4c59-8d20-da34d5cd9d86)
![20240117-184137](https://github.com/apache/rocketmq/assets/113001598/b9b06b25-3212-4003-8fbd-64f5bc02ba01)
![20240117-184142](https://github.com/apache/rocketmq/assets/113001598/e75fe922-e988-44f2-a305-9c12c92e0526)

**Comment by MmringZee** [comment]
Here's the first picture.
![20240117-184124](https://github.com/apache/rocketmq/assets/113001598/153e84bd-5545-48d0-9d04-43a1c2a02c1a)

**Reply by leizhiyuan** [reply]
这是虚拟内存

**Reply by MmringZee** [reply]
但是虚拟内存也意味着rocketmq需要申请到那么多资源才能启动，即使不需要真实地使用。
我现在的场景就是需要在k8s里部署rocketmq，但是因为这个情况我发现我不得不把request设置得很大，这会占用掉很多的集群资源 T.T
我一开始尝试调整jvm参数并把request和limit设置成合理的1、2G甚至2、4G发现都没办法部署一直报OOM。后面在开发环境才发现内存占用量很高这个事情。

**Reply by leizhiyuan** [reply]
你直接在容器里面看 ps aux|grep java 看看jvm参数实际生效的是什么，我们在2C4G的容器里也能跑起来的

**Reply by MmringZee** [reply]
刚看了一下，虚拟内存占了9个多G，实际内存1个多G，很离谱。你们是在rocketmq打成镜像前把那两个启动脚本里的JVM参数改了就ok了吗？没有做其他和内存相关的配置了嘛

**Reply by leizhiyuan** [reply]
没有，虚拟内存是可以大于物理内存的啊
